test/domain_test.rb

```diff
@@ -140,6 +141,15 @@ def test_auth_does_not_bind
     assert user = @domain.user?('user1')
     refute @domain.auth(user, 'foo'), 'Expected user not not bind'
   end
+
+  def test_user_search_returns_first_entry
+    entry = Object.new
+    search_strategy = Object.new
```

Comment:
You could use `mock("Net::LDAP::Entry")` and `mock("UserSearch strategy")` here.